Content.Server/Dragon/DragonSystem.cs

```diff
@@ -403,6 +416,73 @@ private void OnDevourAction(EntityUid uid, DragonComponent component, DragonDevo
             });
         }
 
+        private void UpdateFireBreath(DragonComponent component)
+        {
+            if (component.FireBreathCurrent == null || component.FireBreathCurrent.NextBreath > _timingSystem.CurTime)
+                return;
+
+            if (component.BreathProjectilePrototype == null)
+                return;
+
+            var xform = Transform(component.Owner);
+
+            var random = _randomSystem.NextFloat(-1.0f, 1.0f);
+            var spread = Angle.FromDegrees(25f * random);
+            var angle = component.FireBreathCurrent.MapDirection.ToWorldAngle() + spread;
+            var direction = angle.ToWorldVec();
+
+            var breathSpawn = xform.MapPosition.Offset(direction * 0.8f);
+            var breathProjectileUid = Spawn(component.FireBreathCurrent.BreathPrototype, breathSpawn);
+            ShootBreathProjectile(breathProjectileUid, direction, component.Owner, component.FireBreathCurrent.Speed);
+
+            component.FireBreathCurrent.NextBreath = _timingSystem.CurTime + component.FireBreathCurrent.Delay;
+            component.FireBreathCurrent.BreathsRemaining--;
+
+            if (component.FireBreathCurrent.BreathsRemaining <= 0)
+                component.FireBreathCurrent = null;
+        }
+
+        private void OnDragonBreathFire(EntityUid uid, DragonComponent component,
+            DragonBreatheFireActionEvent args)
+        {
+            if (args.Handled || component.BreatheFireAction == null || component.BreathProjectilePrototype == null)
+                return;
+
+            var dragonXform = Transform(uid);
+            var dragonPos = dragonXform.MapPosition;
+
+            var breathDirection = (args.Target.Position - dragonPos.Position);
+            if (breathDirection == Vector2.Zero || breathDirection == Vector2.NaN)
+                return;
+
+            component.FireBreathCurrent = new DragonFirebreath
+            {
+                NextBreath = TimeSpan.Zero,
+                BreathsRemaining = 5,
+                MapDirection = breathDirection.Normalized,
+                Speed = 5f,
+                Delay = TimeSpan.FromSeconds(0.3f),
+                BreathPrototype = component.BreathProjectilePrototype
+            };
+
+            args.Handled = true;
+
+            if (component.SoundBreathFire != null)
+                _audioSystem.PlayPvs(component.SoundBreathFire, uid, component.SoundBreathFire.Params);
+        }
+
+        private void ShootBreathProjectile(EntityUid uid, Vector2 direction, EntityUid user, float speed)
+        {
+            var physics = EnsureComp<PhysicsComponent>(uid);
+            physics.BodyStatus = BodyStatus.InAir;
+            physics.LinearVelocity = direction * speed;
+
+            var projectile = EnsureComp<ProjectileComponent>(uid);
+            _projectileSystem.SetShooter(projectile, user);
+
+            Transform(uid).WorldRotation = direction.ToWorldAngle();
+        }
+
```

Comment:
You should make the action a toggle to add guncomponent to the dragon as otherwise there will be multiple sets of shooting code to maintain. Master has already had a few changes to this + it will also need updating when pause offsets get merged.